### ISSUE TITLE
Refactor accessibility providers into better separated model

### ISSUE THREAD

**Issue by carlos-zamora** [microsoft/terminal#2120]
# Description of the new feature/enhancement

ScreenInfoUiaProvider and UiaTextRange are in a shared namespace now. PR #2083 did a bit of hacking to get them to work in Windows Terminal. One big example of this is making the HWND optional.

The CORRECT approach should be to do some kind of inheritance model where the ConHost accessibility model uses an HWND and the Windows Terminal accessibility model uses its new method.

For the HWND issue specifically, it might be worth doing this alongside #2103

**Comment by carlos-zamora** [comment]
Another one:
WindowUiaProviderBase pUiaParent shouldn't be nullable in ScreenInfoUiaProvider
